packages/graphql/src/classes/Subgraph.ts

```diff
@@ -103,13 +107,24 @@ export class Subgraph {
         });
     }
 
-    public getReferenceResolvers(nodes: Node[]): IResolvers {
+    public getReferenceResolvers(nodes: Node[], schemaModel: Neo4jGraphQLSchemaModel): IResolvers {
         const resolverMap: IResolvers = {};
 
         const document = mergeTypeDefs(this.typeDefs);
 
         document.definitions.forEach((def) => {
             if (def.kind === Kind.OBJECT_TYPE_DEFINITION) {
+                const entity = schemaModel.getEntity(def.name.value);
+
+                if (schemaModel.isConcreteEntity(entity)) {
```

Comment:
It's an interesting one!

With a getter, we'd have to traverse all the entities in the `concreteEntities` or `compositeEntities` arrays, rather than just getting the entity directly from the `entities` map.

Also, let's imagine a scenario where we want to do something if it's a concrete entity and something else if it's a composite entity. I think it makes more sense to write it like this (or a variation with guard clauses):

```ts
const entity = schemaModel.getEntity(def.name.value);

if (schemaModel.isConcreteEntity(entity)) {
    // Do something that's only possible on a concrete entity
} else if (schemaModel.isCompositeEntity(entity)) {
    // Do something that's only possible on a composite entity
} else {
    // It was neither, so probably shouldn't be handled here
}
```

As opposed to something like this:

```ts
const concreteEntity = schemaModel.getConcreteEntity(def.name.value); // Possibly undefined, or an error?

if (concreteEntity) {
    // Do something that's only possible on a concrete entity
}

const compositeEntity = schemaModel.getCompositeEntity(def.name.value); // Possibly undefined, or an error?

if (compositeEntity) {
    // Do something that's only possible on a concrete entity
}
```

Reply:
Addressed in https://github.com/neo4j/graphql/pull/3284/commits/42ee2d318e9f4653fefd4179aeaf8b9560b83b40